src/coreclr/jit/morphblock.cpp

```diff
@@ -1148,28 +1149,13 @@ GenTree* MorphCopyBlockHelper::CopyFieldByField()
                     // We will spill m_dstAddr (i.e. assign to a temp "BlockOp address local")
                     // no need to clone a new copy as it is only used once
                     //
-                    addrSpill = m_dstAddr; // addrSpill represents the 'm_dstAddr'
+                    addrSpill          = m_dstAddr; // addrSpill represents the 'm_dstAddr'
+                    addrSpillSrcLclNum = m_dstLclNum;
                 }
             }
         }
     }
 
-    // TODO-CQ: this should be based on a more general
-    // "BaseAddress" method, that handles fields of structs, before or after
-    // morphing.
-    if ((addrSpill != nullptr) && addrSpill->OperIs(GT_ADDR))
-    {
-        GenTree* addrSpillOp = addrSpill->AsOp()->gtGetOp1();
-        if (addrSpillOp->IsLocal())
-        {
-            // We will *not* consider this to define the local, but rather have each individual field assign
-            // be a definition.
-            addrSpillOp->gtFlags &= ~(GTF_LIVENESS_MASK);
```

Comment:
This was deleted as it was redundant with the code [above](https://github.com/dotnet/runtime/blob/8f80eed12d4f9846b3300ae97ff077d6c71746d7/src/coreclr/jit/morphblock.cpp#L1123-L1132).